playbook.yml

```diff
@@ -1,18 +1,17 @@
 ---
-- hosts: postgres_cluster
+- hosts: postgres_replication
   vars:
-    create_users: []
-    # - user: test
-    #   comment: Test user
-    #   generate_keys: true
-    #   pass: test
-    #   preseed_user: false
-    #   state: present
-    #   sudo: false
-    #   system_account: false
+    postgresql_config: true
+    postgresql_enable_replication: true
+    postgresql_listen_addresses:
+      - "*"
+    postgresql_replication:
+      interface: enp0s8
+      master: "{{ groups[postgresql_replication_group][0] }}"
+      user: repluser
+    postgresql_replication_group: postgres_replication
     pri_domain_name: test.vagrant.local
+
   roles:
-    - role: ansible-users
-      when: create_users is defined
     - role: ansible-postgresql
-  tasks:
+      become: true
```

Comment:
This may be a "BC-break" for you as well. I removed all the `become: true` in task files, as they all seemed to require it, I moved it to the "front" (to the example).

Reply:
Yeah this is a habit I have gotten myself into by only adding `become: true` to the actual tasks that require it. This way you don't have to execute the whole role as `become: true` which may not be required for some tasks. But as you mentioned, most of the tasks for this role require it.